File: src/Tests.cs/WormholeTests.cs
using System.Collections.Generic;
using FluentAssertions;
using NUnit.Framework;
using TLK.LoveDeathFSharp.Kernel;

namespace TLK.LoveDeathFSharp.Tests
{
    internal class WormholeTests
    {
        private readonly Spaceship blueGoose = new Spaceship("Blue Goose");
        private readonly SpaceStation samulakiStation = new SpaceStation("Samulaki Station");
        private readonly Spaceship serenity = new Spaceship("Serenity");
        private readonly SpaceStation babylon5 = new SpaceStation("Babylon 5");
        private readonly Spaceship ussVoyager = new Spaceship("USS Voyager");

        [Test]
        public void WormholeShouldDisplaceBlueGooseFromLocalBubbleToSchedarSector()
        {
            // arrange
            var localBubble = new List<Spacecraft> { blueGoose, serenity, babylon5, ussVoyager };
            var schedarSector = new List<Spacecraft> { samulakiStation };
            var wormhole = new Wormhole(localBubble, schedarSector);

            // act
            wormhole.Displace(blueGoose);

            // assert
            localBubble.Should().NotContain(blueGoose);
            schedarSector.Should().Contain(blueGoose);
        }
    }
}

File: src/Tests.fs/WormholeTests.fs
module TLK.LoveDeathFSharp.Tests.Wormhole

open NUnit.Framework
open FsUnit.TopLevelOperators
open TLK.LoveDeathFSharp.Kernel
open System.Linq

let ship = Spaceship
let station = SpaceStation
let spaceSector = Enumerable.ToList<Spacecraft>

let (blueGoose, samulakiStation, serenity, babylon5, ussVoyager) =
    (ship "Blue Goose", station "Samulaki Station", ship "Serenity", station "Babylon 5", ship "USS Voyager")

let displace ship origin destination =
    let wormhole = (origin, destination) |> Wormhole
    ship |> wormhole.Displace

[<Test>]
let ``wormhole should displace Blue Goose from Local Bubble to Schedar Sector`` () =
    let localBubble = spaceSector [ blueGoose; serenity; babylon5; ussVoyager ]
    let schedarSector = spaceSector [ samulakiStation ]

    localBubble |> displace blueGoose <| schedarSector

    localBubble |> should not' (contain blueGoose)
    schedarSector |> should contain blueGoose
